packages/@internationalized/date/src/queries.ts

```diff
@@ -180,6 +180,25 @@ export function getMinimumDayInMonth(date: AnyCalendarDate) {
   return 1;
 }
 
+/** Returns an array of the start dates of each month within the specified date interval. */
+export function eachMonthOfInterval(start: DateValue, end: DateValue): DateValue[] {
+  const startDate = startOfMonth(start).set({hour: 0, minute: 0});
+  const endDate = endOfMonth(end).set({hour: 0, minute: 0});
+
+  if (startDate.compare(endDate) > 0) {
+    throw new Error('The start date must be before the end date.');
+  }
+
+  function collectDates(currentDate: DateValue, endDate: DateValue): DateValue[] {
+    if (currentDate.compare(endDate) > 0) {
+      return [];
+    }
+    return [currentDate, ...collectDates(startOfMonth(currentDate.add({months: 1})), endDate)];
+  }
+
+  return collectDates(startDate.add({months: 1}), endDate);
```

Comment:
Why did you choose to exclude the start month in the interval?

I was looking at the [date-fns docs](https://date-fns.org/v3.6.0/docs/eachMonthOfInterval#examples) and it seems like the interval is inclusive of the start month so I was just curious about this choice. Not sure if I'm maybe just missing something.

Reply:
@yihuiliao 
Thank you for your comment!

I forgot that the JavaScript Date type starts counting months from 0, which led to an incorrect implementation.
I have fixed it to include the start month in the result, so I would appreciate it if you could check it again.

[fix: include the start month in the result](https://github.com/adobe/react-spectrum/pull/6647/commits/a12e1b349a138ff6ece3727447535d9bac3de551)